core/src/main/java/org/mule/runtime/core/api/retry/RetryPolicyTemplate.java

```diff
@@ -29,4 +34,11 @@
   void setNotifier(RetryNotifier retryNotifier);
 
   RetryContext execute(RetryCallback callback, Executor workManager) throws Exception;
+
+  default <T> Publisher<T> applyPolicy(Publisher<T> publisher,
```

Comment:
this is following the same pattern as when blocking retry is performed. The user sees the template but the actual retry logic resides in the policy, and a new instance of that policy is created for each retryable task. This allows the user to have a single way of configuring reconnection, and then internally we decide when to do it in a blocking or non blocking fashion.